scipy/optimize/tests/test_trustregion_krylov.py

```diff
@@ -0,0 +1,103 @@
+"""
+Unit tests for Krylov space trust-region subproblem solver.
+
+To run it in its simplest form::
+  nosetests test_optimize.py
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize._trlib import (TRLIBQuadraticSubproblem)
+from numpy.testing import (TestCase, assert_, assert_array_equal,
+                           assert_almost_equal,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_less, run_module_suite)
+
+KrylovQP = TRLIBQuadraticSubproblem(tol_rel_i=1e-8, tol_rel_b=1e-6)
+
+class TestKrylovQuadraticSubproblem(TestCase):
+
+    def test_for_the_easy_case(self):
+
+        # `H` is chosen such that `g` is not orthogonal to the
+        # eigenvector associated with the smallest eigenvalue.
+        H = np.array([[1.0, 0.0, 4.0],
+             [0.0, 2.0, 0.0],
+             [4.0, 0.0, 3.0]])
+        g = np.array([5.0, 0.0, 4.0])
```

Comment:
i think it would help the visualisation to align the matrix ``H``.

```   
H = np.array([[1.0, 0.0, 4.0],
              [0.0, 2.0, 0.0],
              [4.0, 0.0, 3.0]])
```